src/context/read-mode.tsx

```diff
@@ -0,0 +1,252 @@
+    /* eslint-disable @typescript-eslint/no-non-null-assertion */
+    @checkServer
+    migrateReadDrive(id: string, options: RemoteDriveOptions) {
+        return this.server!.migrateReadDrive(id, options);
+    }
+
+    @checkServer
+    addReadDrive(url: string, options?: RemoteDriveOptions) {
+        return this.server!.addReadDrive(url, options);
+    }
+
+    @checkServer
+    getReadDrives(): Promise<string[]> {
+        return this.server!.getReadDrives();
+    }
+
+    @checkServer
+    getReadDriveBySlug(
+        slug: string,
+    ): Promise<ReadDrive | ReadDriveSlugNotFoundError> {
+        return this.server!.getReadDriveBySlug(slug);
+    }
+
+    @checkServer
+    getReadDrive(id: string): Promise<ReadDrive | ReadDriveNotFoundError> {
+        return this.server!.getReadDrive(id);
+    }
+
+    @checkServer
+    getReadDriveContext(
+        id: string,
+    ): Promise<ReadDriveContext | ReadDriveNotFoundError> {
+        return this.server!.getReadDriveContext(id);
+    }
+
+    @checkServer
+    fetchDrive(id: string): Promise<ReadDrive | ReadDriveNotFoundError> {
+        return this.server!.fetchDrive(id);
+    }
+
+    @checkServer
+    fetchDocument<D extends Document>(
+        driveId: string,
+        documentId: string,
+        documentType: DocumentModel<
+            InferDocumentState<D>,
+            InferDocumentOperation<D>,
+            InferDocumentLocalState<D>
+        >['documentModel']['id'],
+    ): Promise<
+        | Document<
+              InferDocumentState<D>,
+              InferDocumentOperation<D>,
+              InferDocumentLocalState<D>
+          >
+        | DocumentModelNotFoundError
+        | ReadDriveNotFoundError
+        | ReadDocumentNotFoundError
+    > {
+        return this.server!.fetchDocument<D>(driveId, documentId, documentType);
+    }
+
+    @checkServer
+    deleteReadDrive(id: string): Promise<ReadDriveNotFoundError | undefined> {
+        return this.server!.deleteReadDrive(id);
+    }
+
+    @checkServer
+    onReadDrivesUpdate(
+        listener: ReadDrivesListener,
+    ): Promise<ReadDrivesListenerUnsubscribe> {
+        return this.server!.onReadDrivesUpdate(listener);
+    }
+
+    /* eslint-enable @typescript-eslint/no-non-null-assertion */
+}
+
+const ReadModeInstance = new ReadModeContextImpl(DefaultDocumentDriveServer);
+
+export const ReadModeContext = createContext<IReadModeContext>({
+    ...(ReadModeInstance as Omit<IReadModeContext, 'readDrives'>),
+    readDrives: [],
+});
+
+export interface ReadModeContextProviderProps {
+    children: ReactNode;
+}
+
+async function getReadDrives(
+    instance: ReadModeContextImpl,
+): Promise<ReadDrive[]> {
+    const driveIds = await instance.getReadDrives();
+    const drives = await Promise.all(
+        driveIds.map(id => instance.getReadDrive(id)),
+    );
+    return drives.filter(
+        drive => !(drive instanceof ReadDriveNotFoundError),
+    ) as ReadDrive[];
+}
+
+export const ReadModeContextProvider: FC<
```

Comment:
The Provider integrates the Read mode into the React App.
Turns the readDrives into React state and changes the default drives to be read drives or normal sync drives according to the user permissions.